topi/python/topi/image/resize.py

```diff
@@ -210,3 +210,185 @@ def _bicubic(*indices):
+                                                  value))
+
+        # use extrapolation_value if in_x is out of boundary
+        out = tvm.if_then_else(in_x < 0,
+                               extrapolation_value,
+                               tvm.if_then_else(in_x > image_width - 1,
+                                                extrapolation_value,
+                                                out_y))
+        return _cast_output(out)
+
+
+    # Bilinear helper functions and computation.
+    def _lerp(A, B, t):
+        return A * (1.0 - t) + B * t
+
+    def _bilinear(*indices):
+        n, c, y, x, cc = _get_indices(*indices)
+        box_idx = box_indices(n)
+
+        y1, x1 = boxes(n, 0), boxes(n, 1)
+        y2, x2 = boxes(n, 2), boxes(n, 3)
+
+        in_h = (image_height - 1) * (y2 - y1)
+        in_w = (image_width - 1) * (x2 - x1)
+        h_scale = tvm.div(in_h, target_h - 1)
+        w_scale = tvm.div(in_w, target_w - 1)
+
+        in_y = y1 * (image_height - 1) + h_scale * y
+        in_x = x1 * (image_width - 1) + w_scale * x
+
+        top_y_index = tvm.floor(in_y).astype('int32')
+        bottom_y_index = tvm.ceil(in_y).astype('int32')
+        y_lerp = in_y - top_y_index
+
+        left_x_index = tvm.floor(in_x)
+        right_x_index = tvm.ceil(in_x)
+        x_lerp = in_x - left_x_index
+
+        top_left = _get_pixel(box_idx, c, top_y_index, left_x_index, cc)
+        top_right = _get_pixel(box_idx, c, top_y_index, right_x_index, cc)
+        bottom_left = _get_pixel(box_idx, c, bottom_y_index, left_x_index, cc)
+        bottom_right = _get_pixel(box_idx, c, bottom_y_index, right_x_index, cc)
+
+        top = _lerp(top_left, top_right, x_lerp)
+        bottom = _lerp(bottom_left, bottom_right, x_lerp)
+        value = _lerp(top, bottom, y_lerp)
+
+        # use extrapolation_value if in_y is out of boundary
+        out_y = tvm.if_then_else(in_y < 0,
+                                 extrapolation_value,
+                                 tvm.if_then_else(in_y > image_height - 1,
+                                                  extrapolation_value,
+                                                  value))
+
+        # use extrapolation_value if in_x is out of boundary
+        out = tvm.if_then_else(in_x < 0,
+                               extrapolation_value,
+                               tvm.if_then_else(in_x > image_width - 1,
+                                                extrapolation_value,
+                                                out_y))
+        return _cast_output(out)
+
+    # Determine which interpolation method to use then run it.
+    if method == "nearest_neighbor":
+        compute_func = _nearest_neighbor
+    elif method == "bilinear":
+        compute_func = _bilinear
+    else:
+        raise ValueError('%s method is not supported.' % method)
+
+    return tvm.compute(output_shape, compute_func, name='crop_and_resize', tag=tag.INJECTIVE)
```

Comment:
A lot of this code is heavily duplicated from the regular resize function. It would make maintenance much easier if we refactored a little so that both resize and resize_and_crop called into the same nearest_neighbor, bilinear, and bicubic kernels. Maybe we can define more general helper functions that will work in both cases.